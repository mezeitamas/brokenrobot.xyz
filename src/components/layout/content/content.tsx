import React from 'react';
import type { FunctionComponent, ReactElement, PropsWithChildren } from 'react';

const Content: FunctionComponent<PropsWithChildren> = ({ children }): ReactElement => {
    return <main className="prose max-w-none md:prose-lg lg:prose-xl">{children}</main>;
};

export { Content };
