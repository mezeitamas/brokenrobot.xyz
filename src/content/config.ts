import { defineCollection, z } from 'astro:content';

const blog = defineCollection({
    schema: ({ image }) =>
        z.object({
            title: z.string(),
            excerpt: z.string(),
            publishDate: z
                .string()
                .or(z.date())
                .transform((val) => new Date(val)),
            heroImage: image(),
            tags: z.array(z.string())
        })
});

export const collections = { blog };
