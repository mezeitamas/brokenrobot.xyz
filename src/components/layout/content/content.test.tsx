import React from 'react';
import type { PropsWithChildren } from 'react';

import { create } from 'react-test-renderer';

import { Content } from './content';

describe('<Content />', () => {
    test('should render', () => {
        // Given
        const props: PropsWithChildren = {
            children: <div>Content</div>
        };

        // When
        const component = create(<Content {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
