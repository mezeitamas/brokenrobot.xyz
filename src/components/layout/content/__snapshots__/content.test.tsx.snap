// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render 1`] = `
<main
  className="prose max-w-none md:prose-lg lg:prose-xl"
>
  <div>
    Content
  </div>
</main>
`;
